### PR TITLE
Map layer button fixes

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
This PR proposes the following fixes to the map layer selector:
- Same style and right margin as the zoom controls.
- Expanded layer selector contents appears above the zoom controls.
- Avoid most accidental activations of the first layer when opening the layer selector using touchscreens.

### CODE REVIEW

File: packages/base-map/src/index.tsx
Comment (miles-grant-ibigroup) [inline packages/base-map/src/index.tsx:163]: I'm not longer able to open the layer selector when using a device without a mouse. Am I doing something wrong?

File: packages/base-map/src/styled.ts
Comment (miles-grant-ibigroup) [inline packages/base-map/src/styled.ts:76]: Why was this removed? I understand matching the zoom buttons but those are tighter and also have a box shadow

File: packages/base-map/src/styled.ts
Comment (miles-grant-ibigroup) [inline packages/base-map/src/styled.ts:99]: This is not happening for me

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
I'd like a cleanup to be this easy, but I'm not sure it's fully working...

File: packages/base-map/src/styled.ts
Comment (binh-dam-ibigroup) [reply packages/base-map/src/styled.ts:99]: Which device(s) are you using?

Comment (binh-dam-ibigroup) [reply packages/base-map/src/styled.ts:99]: *device(s) and browser(s)

Comment (miles-grant-ibigroup) [reply packages/base-map/src/styled.ts:99]: Had the same issue occur on Firefox for Android, Firefox for Mac, and Safari for Mac

File: packages/base-map/src/styled.ts
Comment (binh-dam-ibigroup) [reply packages/base-map/src/styled.ts:76]: I am such a non-believer.

Comment (binh-dam-ibigroup) [reply packages/base-map/src/styled.ts:76]: How about we keep the previous box shadow on hover?

File: packages/base-map/src/index.tsx
Comment (binh-dam-ibigroup) [reply packages/base-map/src/index.tsx:163]: That was a mistake. Reinstated in https://github.com/opentripplanner/otp-ui/pull/578/commits/8c32bb4c8f950822c2ff724d31e0efaf3d25db14.

File: packages/base-map/src/styled.ts
Comment (binh-dam-ibigroup) [reply packages/base-map/src/styled.ts]: Reverted the hover stuff in https://github.com/opentripplanner/otp-ui/pull/578/commits/8c32bb4c8f950822c2ff724d31e0efaf3d25db14.

File: packages/base-map/src/styled.ts
Comment (miles-grant-ibigroup) [reply packages/base-map/src/styled.ts:76]: That's an acceptable compromise to me!

File: packages/base-map/src/styled.ts
Comment (miles-grant-ibigroup) [inline packages/base-map/src/styled.ts:106]: I *think* this line is needed to override other styles that are applied -- even though it's defined above.

**Review by miles-grant-ibigroup — APPROVED** [review]
Looking alright to me, approved pending that box shadow on hover. I'll leave it to the second reviewer to really double check behavior.

File: packages/base-map/src/styled.ts
Comment (binh-dam-ibigroup) [reply packages/base-map/src/styled.ts:76]: I added the on-hover box shadow in https://github.com/opentripplanner/otp-ui/pull/578/commits/7f0e274f8cc99694d04136d1e934cb5fa189c51b.

File: packages/base-map/src/styled.ts
Comment (binh-dam-ibigroup) [reply packages/base-map/src/styled.ts:106]: `display:block` is still in effect on the client (see screenshot).

image

File: packages/base-map/src/styled.ts
Comment (amy-corson-ibigroup) [inline packages/base-map/src/styled.ts:83]: While we're in here, I'm wondering if we can add a `margin-bottom: 0` here and a `margin-bottom: 5px` on `&.fake-mobile-hover` at line 102? There's a small margin on every list item that's skewing the button: 

image

File: packages/base-map/src/styled.ts
Comment (amy-corson-ibigroup) [reply packages/base-map/src/styled.ts:76]: Can confirm the box shadow looks good on hover!

**Review by amy-corson-ibigroup — APPROVED** [review]
Looks really good! I've added a small suggestion regarding a small existing issue with the mobile button:

File: packages/base-map/src/styled.ts
Comment (binh-dam-ibigroup) [reply packages/base-map/src/styled.ts:83]: I am aware of this glitch. That styling is set by OTP-RR, so I am looking to do a PR there to fix that.

**Pull request merged**
